File: blocklyTest.js
function mathIsPrime(n) {
  if (n == 2 || n == 3) return true;
  if (isNaN(n) || n <= 1 || n % 1 !== 0 || n % 2 === 0 || n % 3 === 0) return false;
  for (var x = 6; x <= Math.sqrt(n) + 1; x += 6) {
    if (n % (x - 1) === 0 || n % (x + 1) === 0) return false;
  }
  return true;
}

let runs = 100;
let totalTime = 0;

for (let i = 0; i < runs; i++) {
  let test = 1;
  const t0 = performance.now();

  for (let count = 0; count < 100000; count++) {
    test = (typeof test === 'number' ? test : 0) + 1;
    if (mathIsPrime(test)) {
      // Commented to avoid clutter
      //console.log(test);
    }
  }

  const t1 = performance.now();
  const elapsed = t1 - t0;
  totalTime += elapsed;
  //console.log(`Run ${i + 1}: ${elapsed.toFixed(2)} ms`);
}

let avgTime = totalTime / runs;
console.log(`\n📊 Average time over ${runs} runs: ${avgTime.toFixed(2)} ms`);

File: BlocklyTest.py
import time
from numbers import Number
import math
import statistics  # för enkel standardavvikelse

def math_isPrime(n):
    if not isinstance(n, Number):
        try:
            n = float(n)
        except:
            return False
    if n == 2 or n == 3:
        return True
    if n <= 1 or n % 1 != 0 or n % 2 == 0 or n % 3 == 0:
        return False
    for x in range(6, int(math.sqrt(n)) + 2, 6):
        if n % (x - 1) == 0 or n % (x + 1) == 0:
            return False
    return True

runs = 100
times = []

for i in range(runs):
    test = 1
    testprimes = 0
    start_time = time.time()

    for count in range(100000):
        test = (test if isinstance(test, Number) else 0) + 1
        if math_isPrime(test):
            testprimes = testprimes + 1
            pass

    elapsed = time.time() - start_time
    times.append(elapsed)

avg_time = sum(times) / runs
std_dev = statistics.stdev(times)  # sample standard deviation
cv_percent = (std_dev / avg_time) * 100

print(f"\n📊 Average time over {runs} runs: {avg_time:.6f} seconds")
print(f"📉 Standard deviation: {std_dev:.6f} seconds")
print(f"📈 CV (%): {cv_percent:.2f}")